addons/common/fnc_synchTeamColors.sqf

```diff
@@ -0,0 +1,27 @@
+/* ----------------------------------------------------------------------------
+Internal Function: CBA_fnc_synchTeamColors
+
+Description:
+    Synchs the team colors every second. Does not need to be called manually.
+
+Parameters:
+    None
+
+Returns:
+    Nothing
+
+Author:
+    BaerMitUmlaut
+---------------------------------------------------------------------------- */
+
+#include "script_component.hpp"
+
+if (leader player == player) then {
+    {
+        if ((assignedTeam _x) != (_x getVariable [QGVAR(synchedTeam), "MAIN"])) then {
+            //Local team != currently synched team, so we need to synchronize them again
+            ["CBA_teamColorChanged", [_x, assignedTeam _x]] call CBA_fnc_globalEvent;
```

Comment:
That's possible too, but would require extension of the PFH (unless there's a CBA event for it? there's no doc for default events and I didn't see any either) and it would ofc have a delay. Without that delay it would be great for triggering a CBA event though.

Either options aren't perfect in my opinion. Choose your poison...